multiapps-controller-client/src/main/java/org/cloudfoundry/multiapps/controller/client/ResilientCloudControllerClient.java

```diff
@@ -397,10 +397,21 @@ public void updateServiceSyslogDrainUrl(String serviceName, String syslogDrainUr
     }
 
     @Override
-    public CloudPackage asyncUploadApplication(String applicationName, Path file, UploadStatusCallback callback,
-                                               Duration initialRequestTimeout) {
-        return executeWithExponentialRetry(timeout -> delegate.asyncUploadApplication(applicationName, file, callback, timeout),
-                                           initialRequestTimeout);
+    public CloudPackage asyncUploadApplicationWithExponentialBackoff(String applicationName, Path file, Duration uploadTimeout) {
+        return executeWithExponentialRetry(timeout -> delegate.asyncUploadApplicationWithExponentialBackoff(applicationName, file,
+                                                                                                            timeout));
+    }
+
+    @Override
+    public CloudPackage asyncUploadApplicationWithExponentialBackoff(String applicationName, Path file, UploadStatusCallback callback,
+                                                                     Duration uploadTimeout) {
+        throw new UnsupportedOperationException("Custom upload timeout must not be used!");
+    }
```

Comment:
This is a bit confusing. If the async upload method in **ResilientCloudControllerClient** shouldn't have a custom upload timeout, why doesn't the method on line 400 also throw an exception, since it has **uploadTimeout** as parameter?